src/components/ProductCard.jsx

```diff
@@ -1,153 +1,82 @@
 import React, { Component } from 'react';
-import { Link } from 'react-router-dom';
-import freeShipping from '../img/free-shipping.png';
+import { DetailedProduct, BtnAddToCart } from './';
 
 export default class ProductCard extends Component {
   constructor(props) {
-    super();
+    super(props);
 
     this.state = {
       details: false,
+      quantity: 1,
       productId: props.product.id,
       title: props.product.title,
       price: props.product.price,
       thumbnail: props.product.thumbnail,
-      quantity: 1,
       availableQuantity: props.product.available_quantity,
     };
 
     this.openDetails = this.openDetails.bind(this);
-    this.plainProduct = this.plainProduct.bind(this);
-    this.detailedProduct = this.detailedProduct.bind(this);
     this.addToCart = this.addToCart.bind(this);
-    this.freeShipping = this.freeShipping.bind(this);
-    this.calcShipping = this.calcShipping.bind(this);
   }
 
   addToCart() {
-
-    const obj = {
-      title: this.state.title,
-      price: this.state.price,
-      thumbnail: this.state.thumbnail,
-      quantity: this.state.quantity,
-      availableQuantity: this.state.availableQuantity,
+    const { productId, title, price, thumbnail, quantity, availableQuantity } = this.state;
+    const chosenProduct = {
+      productId,
+      title,
+      price,
+      thumbnail,
+      quantity,
+      availableQuantity,
     };
-
     if (!localStorage.getItem('cart')) {
-      const array = [];
-      array.push(obj);
-      localStorage.clear();
-      localStorage.setItem('cart', JSON.stringify(array));
+      localStorage.setItem('cart', JSON.stringify([chosenProduct]));
+      this.props.updateCartItems(() => [chosenProduct]);
     } else {
-      const save = JSON.parse(localStorage.getItem('cart'));
-      save.push(obj);
+      const cart = JSON.parse(localStorage.getItem('cart'));
+      const indexOfItemInCart = cart.findIndex(product =>
+        product.productId === chosenProduct.productId);
+      if (indexOfItemInCart === -1) {
+        cart.push(chosenProduct)
+      } else {
+        cart[indexOfItemInCart].quantity += chosenProduct.quantity;
+      }
```

Comment:
eu acho que fica melhor deixar uma linha antes e depois de um if, mas fica a teu critério adicionar ou não

```suggestion

      if (indexOfItemInCart === -1) {
        cart.push(chosenProduct)
      } else {
        cart[indexOfItemInCart].quantity += chosenProduct.quantity;
      }
      
```